pdf.go

```
@@ -3,12 +3,110 @@ package docconv
 import (
 	"fmt"
 	"io"
+	"io/ioutil"
 	"log"
+	"os"
 	"os/exec"
+	"path/filepath"
 	"strings"
+	"sync"
 	"time"
 )
 
+var (
+	exts = []string{".jpg", ".tif", ".tiff", ".png"}
+)
+
+var pdfMutex struct {
+	sync.Mutex
+}
+
+func compareExt(ext string, exts []string) bool {
+	for _, e := range exts {
+		if ext == e {
+			return true
+		}
+	}
+	return false
+}
+
+func PDFImages(path string) (string, map[string]string, error) {
+	tmp, err := ioutil.TempDir("/tmp", "tmp-imgs-")
+	if err != nil {
+		log.Println(err)
```

Return the error instead of printing it.